client/src/components/EventPage/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import { Parallax, Row, Col, CardPanel, Icon } from 'react-materialize';
+import moment from 'moment';
+import EventMap from '../EventMap';
+import EventMeta from './EventMeta';
+import { request } from '../../utils';
+
+
+class EventPage extends Component {
+  state = {
+    name: '',
+    description: '',
+    start_date: '',
+    end_date: '',
+    author_id: '',
+    author_name: '',
+    category_id: '',
+    main_image: '',
+    lat: '',
+    long: '',
+    category: '',
+    tags: {},
+  }
+
+  componentDidMount() {
+    request(`/event/${this.props.match.params.id}`).then(data => {
+      if ('event' in data) {
+        for (const key in data.event) {
+          if (this.state.hasOwnProperty(key)) {
+            this.setState({ [key]: data.event[key] });
+          }
+        }
+        this.setState({ category: data.category });
+        this.setState({ tags: data.tags });
+      }
+    })
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.match.params.id !== prevProps.match.params.id) {
+      request(`/event/${this.props.match.params.id}`).then(data => {
+        if ('event' in data) {
+          for (const key in data.event) {
+            if (this.state.hasOwnProperty(key)) {
+              this.setState({ [key]: data.event[key] });
+            }
+          }
+          this.setState({ category: data.category });
+          this.setState({ tags: data.tags });
+        }
+      })
+    }
+  }
+
+  renderTimeString() {
+    const start_day = moment(this.state.start_date).format('MMMM D, YYYY');
+    if (!this.state.end_date) {
+      return <p className="flow-text"><Icon>date_range</Icon>{start_day}<Icon>access_time</Icon>{moment(this.state.start_date).format('hh:mm')}</p>
+    }
+
+    if (start_day === moment(this.state.end_date).format('MMMM D, YYYY')) {
+      return <p className="flow-text"><Icon>date_range</Icon>{start_day}<Icon>access_time</Icon>{moment(this.state.start_date).format('hh:mm') + " - " +
```

Comment:
In my opinion, it's a bit hard to read, maybe need write  every <Icon> like this ->

```
<Icon>
    some text
</Icon>
```

and on 58line